File: 151-reverse-words-in-a-string/reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
    s = s.split(' ');
    let res = []
    for(let i=s.length-1;i>=0;i--){
        if(s[i]!==''){
            res.push(s[i]);
        }
    }
    return res.join(' ')
};

 // s = s.trim()
    // let left = 0
    // let right = s.length - 1
    // let temp = ''
    // let ans = ''
    // while (left <= right) {
    //     let ch = s[left]
    //     if (ch !== ' ') {
    //         temp += ch
    //     } else if (ch === ' ') {
    //         if (ans !== '' && temp !== '') {
    //             ans = temp + ' ' + ans
    //         } else if(ans === ''){
    //             ans = temp
    //         }
    //         temp = ''
    //     }
    //     left++
    // }
    // if (temp !== '') {
    //     if (ans !== '') ans = temp + ' ' + ans
    //     else ans = temp
    //     return ans
    // }

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] strs = s.split(" ");
        StringBuilder res = new StringBuilder();
        String pre = "";
        for (int i = strs.length - 1; i >= 0; i--) {
            if (!strs[i].isEmpty()) {
                res.append(pre);
                pre = " ";
                res.append(strs[i]);
            }
        }
        return res.toString();
    }
}